### PR TITLE
implement < <= > >= for int4 vectors

### PR DESCRIPTION
**Pull request by WalterBright**
Will do other vector types later.

Most of this is just rewriting the AST so everything is in terms of the < operation. The result is 0 for false and ~0 for true for each element of the int4 result value.

### CODE REVIEW

File: compiler/src/dmd/e2ir.d [hunk 1]
```diff
@@ -1819,6 +1819,57 @@ elem* toElem(Expression e, IRState *irs)
...
+            }
+
+            elem* e1 = toElem(ce.e1, irs);
+            elem* e2 = toElem(ce.e2, irs);
+
+            elem* ex;
+            if (swap)
+            {
+                // put side effects of e1 into ex
+                if (el_sideeffect(e1) && e2.Eoper != OPconst)
```
Comment (dkorpel) [inline compiler/src/dmd/e2ir.d:1844]: There's no test for comparisons with side effects in the operands

**Comment by dkorpel** [comment]
The operators should be added to this table: https://dlang.org/spec/simd.html#x86_64_vec

**Comment by WalterBright** [comment]
> The operators should be added to this table

Yes, but I haven't finished yet supporting all the comparison operations. They only work for int4 vectors at the moment.

File: compiler/src/dmd/e2ir.d (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/e2ir.d:1844]: there is now

**Review by dkorpel — APPROVED** [review]

**Pull request merged**